### PR TITLE
STORM-1097. Compile logical plans to Java source code.

### PR DESCRIPTION
**Pull request by haohui**
This PR implements the compilers that compile logical plans from Calcite to Java source code. It also removes the compiler implemented in native code per discussions in https://issues.apache.org/jira/browse/STORM-1105?focusedCommentId=14970164&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14970164.

### CODE REVIEW

File: external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/ExprCompiler.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    private String compile(ExprCompiler compiler, RexCall call) {
+      SqlOperator op = call.getOperator();
+      CallExprPrinter printer = translators.get(op);
+      if (printer == null) {
+        throw new UnsupportedOperationException();
+      } else {
+        return printer.translate(compiler, call);
+      }
+    }
+    private Map.Entry<SqlOperator, CallExprPrinter> infixBinary
```
Comment (harshach) [inline external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/ExprCompiler.java]: new line here.

File: external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/ExprCompiler.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    pw.append(
+        String.format("%s %s = (%s)(%s);\n", typeName, name, typeName, definition));
+    return name;
+  }
+
+  private interface CallExprPrinter {
+    String translate(ExprCompiler compiler, RexCall call);
+  }
+
+  private static class ImpTable {
```
Comment (harshach) [inline external/sql/storm-sql-core/src/jvm/org/apache/storm/sql/compiler/ExprCompiler.java]: what is ImpTable can you add some details here.

**Comment by harshach** [comment]
@haohui overall looks good to me. couple of minor nitpicks.

**Comment by haohui** [comment]
Thanks @harshach for the reviews.

ImpTable maps the SQL operators to the classes that understand the SQL operators and generate corresponding Java source code.

I updated the patch and added comments for the class.

**Comment by harshach** [comment]
+1

**Pull request merged**